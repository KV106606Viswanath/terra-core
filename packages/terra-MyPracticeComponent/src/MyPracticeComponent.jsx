import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import classNamesBind from 'classnames/bind';
import ThemeContext from 'terra-theme-context';
import styles from './MyPracticeComponent.module.scss';

const cx = classNamesBind.bind(styles);

const propTypes = {
  /**
   * Content to be displayed as the name
   */
  name: PropTypes.string,
};

const defaultProps = {
  name: 'default',
};

const MyPracticeComponent = ({ name, ...customProps }) => {
  const theme = React.useContext(ThemeContext);
  const MyPracticeComponentClassNames = classNames(
    cx([
      'MyPracticeComponent',
      theme.className,
    ]),
    customProps.className,
  );

  return (<button type="button" {...customProps} className={MyPracticeComponentClassNames}>{name}</button>);
};

MyPracticeComponent.propTypes = propTypes;
MyPracticeComponent.defaultProps = defaultProps;

export default MyPracticeComponent;
