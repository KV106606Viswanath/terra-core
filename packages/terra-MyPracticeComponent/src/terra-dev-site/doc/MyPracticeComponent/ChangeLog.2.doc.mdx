import { Badge } from 'terra-MyPracticeComponent/package.json?dev-site-package';
import ChangeLog from 'terra-MyPracticeComponent/CHANGELOG.md';

<Badge />

<ChangeLog />
