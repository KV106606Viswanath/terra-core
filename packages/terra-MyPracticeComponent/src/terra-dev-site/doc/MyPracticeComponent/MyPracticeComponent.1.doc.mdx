import { Badge } from 'terra-MyPracticeComponent/package.json?dev-site-package';

import DefaultMyPracticeComponent from '../example/DefaultMyPracticeComponent?dev-site-example';

import PropsTable from 'terra-MyPracticeComponent/src/MyPracticeComponent?dev-site-props-table';

<Badge />

# Terra Mypracticecomponent

{insert description}

## Getting Started

- Install with [npmjs](https://www.npmjs.com):
  - `npm install terra-MyPracticeComponent`

## Usage

```jsx
import React from 'react';
import MyPracticeComponent from 'terra-MyPracticeComponent';

<MyPracticeComponent {props...} />
```

## Component Features

* [Cross-Browser Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#cross-browser-support)
* [LTR/RTL Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#ltr--rtl-support)

## Examples
<DefaultMyPracticeComponent />

## Divider Props
<PropsTable />
