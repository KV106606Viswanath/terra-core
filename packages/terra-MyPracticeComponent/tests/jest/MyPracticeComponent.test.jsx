import React from 'react';
import MyPracticeComponent from '../../src/MyPracticeComponent';

describe('MyPracticeComponent', () => {
  const defaultRender = <MyPracticeComponent />;

  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });

  // Prop Tests
  it('should use the default value when no value is given', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.find('.MyPracticeComponent').text()).toEqual('default');
  });

  // Structure Tests
  it('should have the class MyPracticeComponent', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.prop('className')).toContain('MyPracticeComponent');
  });
});
