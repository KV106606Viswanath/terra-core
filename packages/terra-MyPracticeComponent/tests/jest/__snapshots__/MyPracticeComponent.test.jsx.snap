// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPracticeComponent should render a default component 1`] = `
<button
  className="MyPracticeComponent"
>
  default
</button>
`;
